community/kernel/src/main/java/org/neo4j/kernel/recovery/Recovery.java

```diff
@@ -117,14 +126,35 @@ public void init() throws Throwable
             }
             else
             {
-                monitor.failToRecoverTransactionsAfterPosition( t, recoveryFromPosition );
-                recoveryToPosition = recoveryFromPosition;
+                monitor.failToRecoverTransactionsAfterPosition( t, recoveryPosition );
+                recoveryToPosition = recoveryPosition;
             }
         }
+        progressReporter.completed();
         logPruner.prune( recoveryToPosition );
 
         recoveryService.transactionsRecovered( lastTransaction, recoveryToPosition );
         startupStatistics.setNumberOfRecoveredTransactions( numberOfRecoveredTransactions );
         monitor.recoveryCompleted( numberOfRecoveredTransactions );
     }
+
+    private void initProgressReporter( RecoveryStartInformation recoveryStartInformation,
+            CommittedTransactionRepresentation lastReversedTransaction )
+    {
+        long numberOfTransactionToRecover =
+                getNumberOfTransactionToRecover( recoveryStartInformation, lastReversedTransaction );
+        progressReporter.start( numberOfTransactionToRecover << 1 );
```

Comment:
This is due to both reverse and forward recovery, right? Please comment that here and also the `<< 1` is unnecessary, can might as well do `* 2` for better readability.

Reply:
sure, I can do * 2 if you think it's more readable, correct since we will process each transaction twice we need to double expected number of transaction to process.